### PR TITLE
Add script for summarizing metrics and a random agent

### PR DESCRIPTION
**Pull request by yvsriram**
## Motivation and Context

- Processes the metric files to report useful metrics
- Random agent taking random actions

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] I ran `tests/hw_manual_test.py` on hardware and observed that robot behavior looks normal.

### CODE REVIEW

File: projects/habitat_ovmm/README.md [hunk 1]
```diff
@@ -121,12 +121,16 @@ python -u -m habitat_baselines.run \
 ```
 
 
-## Running evaluations
+# Running evaluations
```
Comment (cpaxton) [inline projects/habitat_ovmm/README.md:124]: Why is this changed?

**Review by cpaxton — APPROVED** [review]
LGTM, one question about README formatting

File: projects/habitat_ovmm/eval_baselines_agent.py [hunk 2]
```diff
@@ -57,6 +58,13 @@ def merge_configs(habitat_config, baseline_config):
         default="projects/habitat_ovmm/configs/agent/hssd_eval.yaml",
         help="Path to config yaml",
     )
+    parser.add_argument(
+        "--agent_type",
+        type=str,
+        default="ovmm",
```
Comment (rpartsey) [inline projects/habitat_ovmm/eval_baselines_agent.py:64]: There is no "ovmm" option in choices.

File: projects/habitat_ovmm/evaluator.py [hunk 3]
```diff
@@ -150,7 +170,8 @@ def stop():
                         **metrics_at_skill_end,
                         **episode_metrics[episode_key],
                     }
-                    episode_metrics[episode_key]["goal_name"] = info["goal_name"]
+                    if "goal_name" in episode_metrics[episode_key]:
```
Comment (rpartsey) [inline projects/habitat_ovmm/evaluator.py:173]: The same question here: 
Why have you added `if "goal_name" in episode_metrics[episode_key]`? There was no such check before (in older evaluator code).

File: projects/habitat_ovmm/metrics_utils.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
+from typing import Optional
```
Comment (rpartsey) [inline projects/habitat_ovmm/metrics_utils.py:1]: Merge main and move this file to `projects/habitat_ovmm/utils`.

File: projects/habitat_ovmm/metrics_utils.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
```
Comment (rpartsey) [inline projects/habitat_ovmm/metrics_utils.py:2]: Add copyright
```
# Copyright (c) Meta Platforms, Inc. and affiliates.
#
# This source code is licensed under the MIT license found in the
# LICENSE file in the root directory of this source tree.
```

File: projects/habitat_ovmm/scripts/summarize_metrics.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
+import argparse
```
Comment (rpartsey) [inline projects/habitat_ovmm/scripts/summarize_metrics.py:1]: Add copyright
```
# Copyright (c) Meta Platforms, Inc. and affiliates.
#
# This source code is licensed under the MIT license found in the
# LICENSE file in the root directory of this source tree.
```

File: src/home_robot/home_robot/agent/ovmm_agent/random_agent.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
+from typing import Any, Dict, Tuple
```
Comment (rpartsey) [inline src/home_robot/home_robot/agent/ovmm_agent/random_agent.py:1]: Add copyright
```
# Copyright (c) Meta Platforms, Inc. and affiliates.
#
# This source code is licensed under the MIT license found in the
# LICENSE file in the root directory of this source tree.
```

**Review by rpartsey — CHANGES_REQUESTED** [review]
Please merge main, address my comments below and add all the extra checks you do with episode metrics dict not only do `OVMMEvaluator`'s vectorized eval method, but also to `local_evaluate` and `remote_evaluate`.

File: src/home_robot/home_robot/agent/ovmm_agent/random_agent.py (hunk 7 above)
Comment (yvsriram) [reply src/home_robot/home_robot/agent/ovmm_agent/random_agent.py:6]: Added

File: projects/habitat_ovmm/metrics_utils.py (hunk 5 above)
Comment (yvsriram) [reply projects/habitat_ovmm/metrics_utils.py]: Added

File: projects/habitat_ovmm/metrics_utils.py (hunk 4 above)
Comment (yvsriram) [reply projects/habitat_ovmm/metrics_utils.py]: Done

File: projects/habitat_ovmm/evaluator.py (hunk 3 above)
Comment (yvsriram) [reply projects/habitat_ovmm/evaluator.py:134]: This was being set inside ObjectNavAgent and I am not setting this in RandomAgent

File: projects/habitat_ovmm/evaluator.py [hunk 8]
```diff
@@ -141,7 +160,8 @@ def stop():
                 )
                 if episode_key not in episode_metrics:
                     episode_metrics[episode_key] = {}
-                if info["skill_done"] != "":
+                # Record metrics after each skill finishes. This is useful for debugging.
+                if "skill_done" in info and info["skill_done"] != "":
```
[the remark this answers is not in the document]
Comment (yvsriram) [reply projects/habitat_ovmm/evaluator.py:125]: Because I am not setting these inside RandomAgent

File: projects/habitat_ovmm/evaluator.py [hunk 9]
```diff
@@ -92,6 +94,23 @@ def write_results(self, episode_metrics):
         with open(f"{self.results_dir}/episode_results.json", "w") as f:
             json.dump(episode_metrics, f, indent=4)
 
+    def summarize_metrics(self, episode_metrics: dict) -> dict:
```
[the remark this answers is not in the document]
Comment (yvsriram) [reply projects/habitat_ovmm/evaluator.py]: Changed

File: projects/habitat_ovmm/eval_baselines_agent.py (hunk 2 above)
Comment (yvsriram) [reply projects/habitat_ovmm/eval_baselines_agent.py]: Changed

File: projects/habitat_ovmm/evaluator.py [hunk 10]
```diff
@@ -12,10 +12,12 @@
 from typing import TYPE_CHECKING, Any, Dict, Optional
 
 import numpy as np
+import pandas as pd
 from habitat_baselines.rl.ppo.ppo_trainer import PPOTrainer
 from omegaconf import DictConfig
 from tqdm import tqdm
 from utils.env_utils import create_ovmm_env_fn
+from utils.metrics_utils import get_stats_from_episode_metrics
```
Comment (rpartsey) [inline projects/habitat_ovmm/evaluator.py:20]: There is no `utils.metrics_utils.py` file

File: projects/habitat_ovmm/scripts/summarize_metrics.py [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+import os
+import sys
+from enum import IntEnum, auto
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+
+sys.path.append("projects/habitat_ovmm/")
+from utils.metrics_utils import get_stats_from_episode_metrics
```
Comment (rpartsey) [inline projects/habitat_ovmm/scripts/summarize_metrics.py:17]: There is no `utils.metrics_utils.py` file

**Review by rpartsey — CHANGES_REQUESTED** [review]
`utils.metrics_utils.py` file is missing.

File: projects/habitat_ovmm/scripts/summarize_metrics.py (hunk 11 above)
Comment (yvsriram) [reply projects/habitat_ovmm/scripts/summarize_metrics.py:17]: Added

**Review by rpartsey — APPROVED** [review]
If metrics computation functionality is tested I'm ok with this PR
